mlflow/models/signature.py

```diff
@@ -398,6 +398,11 @@ def _infer_signature_from_type_hints(
     try:
         input_schema = _infer_schema_from_list_type_hint(type_hints.input)
     except InvalidTypeHintException as e:
+        raise MlflowException.invalid_parameter_value(
+            "Remove the invalid type hint from the `predict` signature or use a valid type hint "
+            f"for the model input argument. Error: {e}"
```

Comment:
```suggestion
            "The `predict` function has unsupported type hints for the model input "
            "arguments. Update it to one of supported type hints, or remove type hints "
            "bypass this check.  Error: {e}"
```

nit: 
1. `invalid` -> `unsupported`.
2. Removing type hint is not the best practice in terms of code maintainability; so state it as a short-cut hack.